packages/core/src/input/inputStyle.js

```diff
@@ -22,6 +22,13 @@ export const searchIconCss = () => ({
   left: "8px"
 });
 
+export const clearIconCss = () => ({
+  position: "absolute",
+  top: "8px",
+  right: "8px",
+  cursor: "pointer"
```

Comment:
Is position absolute the right placement?
The text should go under the icon so wouldn't it make more sense to just place the icon after the input?

The problem with the current solution is that you expect a specific height and padding when it's not something you can ensure.
Instead why not use css layout like flex to define something like align-items: center ?